testing/test.bash

```diff
@@ -0,0 +1,42 @@
+script_path=$(dirname $0)
```

Comment:
why do we need this...? (average not a bash user)

Reply:
For the functions called below here? Just means that if we call this file from another directory it doesn't reek havoc on that directory.

```suggestion
script_path=$(dirname -- "$0")
```

Reply:
yeah bash scripts are basically just a bundle of bash commands that are executed in your current session. So if you call the script from the ZP folder it could fuck up the normal build, or call the wrong CMakeLists.txt